GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -180,7 +173,11 @@ public bool TryStart(out string error)
 
             this.backgroundFileSystemTaskRunner.Start();
 
-            this.IsMounted = true;
+            if (!this.fileSystemVirtualizer.TryStart(this, out error))

```

Comment:
We do have some file system calls in the background thread.  Will this cause issues if we start processing the background operations and potentially process one that make a file system call before this TryStart is finished?